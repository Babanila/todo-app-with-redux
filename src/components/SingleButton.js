import React from 'react'

function SingleButton({ btnName, btnClick }) {
  return <button onClick={() => btnClick()}>{btnName}</button>
}

export default SingleButton
