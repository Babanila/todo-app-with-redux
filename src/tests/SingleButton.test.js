import React from 'react'
import { shallow } from 'enzyme'
import SingleButton from '../components/SingleButton'

describe('SingleButton', () => {
  let props, SingleButtonWrapper

  beforeAll(() => {
    props = {
      btnName: 'delete',
      btnClick: jest.fn()
    }

    SingleButtonWrapper = shallow(<SingleButton {...props} />)
  })

  it('should be defined', () => {
    expect(SingleButton).toBeDefined()
  })

  it('should be a function', () => {
    expect(typeof SingleButton).toBe('function')
  })

  it('should renders site title correctly', () => {
    const wrapper = shallow(<SingleButton {...props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the exact number of element in component', () => {
    expect(SingleButtonWrapper.find('button').length).toEqual(1)
  })

  it('should perform operation on button click', () => {
    const mockedButtonClick = jest.fn()
    const wrapper = shallow(<SingleButton {...props} btnClick={mockedButtonClick} />)
    const btn = wrapper.find('button')
    btn.find('button').simulate('click')
    expect(mockedButtonClick.mock.calls.length).toEqual(1)
  })
})
