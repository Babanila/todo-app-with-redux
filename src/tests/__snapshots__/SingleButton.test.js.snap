// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleButton should renders site title correctly 1`] = `
<button
  onClick={[Function]}
>
  delete
</button>
`;
